mongo/src/org/immutables/mongo/bson4gson/GsonCodecs.java

```diff
@@ -150,9 +169,43 @@ public <T> Codec<T> get(Class<T> clazz) {
     };
   }
 
+  /**
+   * Appends a TypeAdapterFactory to allow Gson and Bson adapters to co-exist.
+   * Avoids registering the TypeAdapterFactory if Gson already has factories for BSON types.
+   *
+   * @param gson preconfigured instance
+   * @return {@code gson} configured with delegating type adapter factory.
+   */
+  public static Gson newGsonWithDelegatingTypeAdapterFactory(Gson gson) {
+    if (hasBsonTypeAdapters(gson)) {
```

Comment:
@asereda-gs not too happy with this approach but could not identify a better way to detect if these type adapters were already registered. an alternative I considered was to add another builder method to `RepositorySetup` that allows providing a preconfigured Gson instance. please let me know your thoughts.